### DISCUSSION TITLE
[Feature] Sort smart search results chronologically

### DISCUSSION THREAD

**Discussion by Ynng** [immich-app/immich#8801] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Sort smart search results chronologically like **_App-Which-Must-Not-Be-Named_**.

The current ordering based on relevancy is incomprehensible to humans, with photos taken right after each other of the same subject often separated within the results. This makes it hard to browse through the search results in a coherent manner.

Additionally, the ML models used for smart search is not perfect. I've noticed blurry or blank photos often *peak its interest* and may rank highly in the search results. By sorting the results chronologically, the impact of these would be less noticeable, as they would be distributed throughout the results rather than clustered at the top.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by mmomjian** [comment]
I agree this is a valuable request. This is a duplicate of #8377. Unfortunately this is much harder to implement than it first seems.

**Comment by mws-weekend-projects** [comment]
I implemented this feature as a general UI addition "sort latest/oldest" in this PR https://github.com/immich-app/immich/pull/31572. The family is really happy with the now really usable search results.
